security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -46,6 +46,11 @@ var (
 	// with extra SAN (labels, etc) in data path.
 	ProvCert = env.RegisterStringVar("PROV_CERT", "",
 		"Set to a directory containing provisioned certs, for VMs").Get()
+
+	// OutputKeyCertToDir path is set, it will restore the cert from the signed by the CA
+	OutputKeyCertToDir = env.RegisterStringVar("OUTPUT_CERTS", "",
```

Comment:
Why are we defining from pilot/cmd/pilot-agent/main.go?

Reply:
+1 = leave ( and move ) all env variables to pilot-agent/main.go. 
Let's get https://github.com/istio/istio/pull/25367 merged first !

Reply:
Yes Thanks to costin. it seems a large change. Hopes to catch up with ddl, hope to resolve it this week if not will create a issue in the next release focus on the refactor environment as there are many places use same environment from the main.go